src/lib/commandBarConfigs/validators.ts

```diff
@@ -153,3 +153,57 @@ export const loftValidator = async ({
     return 'Unable to loft with selected sketches'
   }
 }
+
+export const shellValidator = async ({
+  data,
+}: {
+  data: { selection: Selections }
+}): Promise<boolean | string> => {
+  if (!isSelections(data.selection)) {
+    return 'Unable to shell, selections are missing'
+  }
+
+  // No validation on the faces, filtering is done upstream and we have the dry run validation just below
+  const face_ids = data.selection.graphSelections.flatMap((s) =>
+    s.artifact ? s.artifact.id : []
+  )
+
+  // We don't have the concept of solid3ds in TS yet.
+  // So we're listing out the sweeps as if they were solids and taking the first one, just like in Rust for Shell:
+  // https://github.com/KittyCAD/modeling-app/blob/e61fff115b9fa94aaace6307b1842cc15d41655e/src/wasm-lib/kcl/src/std/shell.rs#L237-L238
+  // TODO: This is one cheap way to make sketch-on-face supported now but will likely fail multiple solids
+  const object_id = engineCommandManager.artifactGraph
+    .values()
+    .find((v) => v.type === 'sweep')?.pathId
+
+  if (!object_id) {
+    return "Unable to shell, couldn't find the solid"
```

Comment:
Don't want to return Promise.reject?

Reply:
No, the arg configuration for `validation` takes a function that looks like

```
async function validator ({context, data})  : string | boolean {} 
```

If you return `true` passes validation
If you return `false` fails validation but you don't get an error message
If you return `string` it fails validation but you get to see that message within the failure case.


This function is implemented correctly

Reply:
https://github.com/KittyCAD/modeling-app/blob/842ef5ede981bf9f8394e56c9feef0106117bfc7/src/machines/commandBarMachine.ts#L290

Here is the usage of the validator function

Reply:
Now that I think of this I could have probably have thrown a new Error object with a string instead of saying "success" and returning a string with an error message...

It really isn't any error in the sense of the JS code.. I just need to know if an operation failed or passed then why it failed or passed.

The JS code is just proxying the engine result.

Reply:
Yup, thanks for the added context @nadr0!

Reply:
Right I was suggesting changing the type x) All good